source/fundamentals/auth.txt

```diff
@@ -0,0 +1,384 @@
+tab below for instructions and sample code for specifying this authentication
+mechanism:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Connection String
+
+      To specify the ``MONGODB-AWS`` authentication mechanism using a
+      connection string, assign the ``authMechanism`` parameter the value
+      ``MONGODB-AWS`` in your connection string. Your code to instantiate
+      a ``MongoClient`` should resemble the following:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS");
+
+      If you need to specify an AWS session token, include it in the
+      ``authMechanismProperties`` parameter as follows using the format
+      ``AWS_SESSION_TOKEN:<awsSessionToken>``. Your code to instantiate
+      a ``MongoClient`` with a session token should resemble the following:
+
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>&authMechanism=MONGODB-AWS&authMechanismProperties=AWS_SESSION_TOKEN:<awsSessionToken>");
+
+
+   .. tab::
+      :tabid: MongoCredential
+
+      To specify the default authentication mechanism using the
+      ``MongoCredential`` class, use the
+      :java-docs:`createAwsCredential() </apidocs/mongodb-driver-core/com/mongodb/MongoCredential.html#createAwsCredential(java.lang.String,char%5B%5D)>`
+      method. Your code to instantiate a ``MongoClient`` should resemble the
+      following:
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-aws.rst
+
+      If you need to specify an AWS session token, use the
+      :java-docs:`applyConnectionString() </apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyConnectionString(com.mongodb.ConnectionString)>`
+      method to specify a connection string that specifies the
+      ``authMechanism`` and ``AWS_SESSION_TOKEN`` parameter values.
+
+      .. include:: /includes/fundamentals/code-snippets/auth-credentials-aws-session.rst
+
+.. _x509-auth-mechanism:
+
+``X.509``
+~~~~~~~~~
+
+The ``X.509`` authentication mechanism uses
+:wikipedia:`TLS <Transport_Layer_Security>` with X.509 certificates to
+authenticate your user, identified by the relative distinguished names
+(RDNs) of your client certificate.
+
+To retrieve the X.509 certificate *subject* which is used as your
+*username* for authentication, see the MongoDB server :manual:`X.509
+tutorial </tutorial/configure-x509-client-authentication/#add-x-509-certificate-subject-as-a-user>`
+for more information.
+
+The following code snippets show how to specify the connection mechanism,
+using the following placeholders:
+
+* ``username`` - your MongoDB username, :manual:`derived from the certificate subject </tutorial/configure-x509-client-authentication/#add-x-509-certificate-subject-as-a-user>`
```

Comment:
As of MongoDB 3.4, it's no longer required (or recommended) to explicitly provide the username.  The server will pull it out of the certificate for you.  You can just use `MongoCredential.createMongoX509Credential()` programmatically, and just leave out the username on the connection string.  It's much easier that way!

Reply:
Awesome, I'll update the documentation to describe the easier way.